app/models/metric/ci_mixin/state_finders.rb

```diff
@@ -34,18 +48,32 @@ def preload_vim_performance_state_for_ts_iso8601(conditions = {})
 
   # Use VimPerformanceState to populate the scopes with the values from a particular point in time
   # using @last_vps_ts to ensure we don't load at one time and then use at another
-  def vim_performance_state_association(ts, assoc)
-    if assoc.to_s == "miq_regions"
+  def vim_performance_state_association(ts, association_name)
+    if association_name.to_s == "miq_regions"
       return respond_to?(:miq_regions) ? miq_regions : []
     end
 
     if !defined?(@last_vps_ts) || @last_vps_ts != ts
       @last_vps_ts = ts
-      public_send(assoc).reset
-
-      MiqPreloader.preload(self, assoc, vim_performance_state_for_ts(ts).public_send(assoc).load)
+      # we are using a different timetamp
+      # clear out vps related associations
+      class_associations = self.class.reflections.keys.map(&:to_sym)
+      VimPerformanceState::ASSOCIATIONS.each do |vps_assoc|
+        association(vps_assoc).reset if class_associations.include?(vps_assoc)
+      end
     end
 
-    public_send(assoc)
+    association = association(association_name.to_sym)
```

Comment:
Rails can handle both, so we don't need to be explicit here

```suggestion
    association = association(association_name)
```

Reply:
Also, it's weird to me to shadow a method name into a local variable with the same name.  Perhaps this can be `assoc`? (yes, I don't like abbreviations, but in this case it seems warranted to avoid shadowing)